commit b0bff3202f25949a529a19e5c982e040d0505350 (refs/changes/30/3830/2)
Author: Robert Bogart <robert.bogart@joyent.com>
Date:   2018-04-19T04:53:12+00:00 (1 year, 6 months ago)
    
    MANTA-2909 muskie logs incorrect bytesTransferred for some failed, streaming uploads
    Reviewed by: Jordan Hendricks <jordan.hendricks@joyent.com>
    Reviewed by: Jan Wyszynski <jan.wyszynski@joyent.com>
    Approved by: Kody A Kantor <kody.kantor@joyent.com>

diff --git a/lib/obj.js b/lib/obj.js
index 59993b3..173fee2 100644
--- a/lib/obj.js
+++ b/lib/obj.js
@@ -448,11 +448,21 @@ function sharkStreams(req, res, next) {
         return;
     }
 
+    /*
+     * While in the process of streaming the object out to multiple sharks, if a
+     * failure is experienced on one stream, we will essentially treat it as an
+     * overall failure and abandon the process of streaming this object to all
+     * sharks involved.  Note that `next_err()' is wrapped in the `once()'
+     * method because we need only respond to a failure event once.
+     */
     var next_err = once(function _next_err(err) {
         req.log.debug({
             err: err
         }, 'abandoning request');
 
+        /* Record the number of bytes that we transferred. */
+        req._size = check.bytes;
+
         req.removeListener('end', onEnd);
         req.removeListener('error', next_err);
 
